### ISSUE TITLE
Change to default number of variables (columns)  and rows (cases) displayed?

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#4918]
We currently display 1000 rows by 30 columns. 

The (climatic) data set I used today had 52 going up to 54 columns, so not all were displayed (except on my machine where I had changed the default to 60.  

If climatic data has the initial layout with one column for each day-of-the-month there will always be more than 30 columns.  This messed me up today, because x26 was non-numeric.  This variable was not visible with the default layout.  Hence it was difficult to make the change.

In the current defaults I have no problem with the default number of rows.  Often they are many more than 1000.  And you can see them all with the R viewer. 

I wonder if the number of columns (variables) could be increased, say to 100.  I have now noticed that the R-viewer does obey the filter - which is great.  But it is only a viewer.  In the front end we can make changes - but we need to see them to make the changes.

**Comment by dannyparsons** [comment]
There's two considerations:

1. Impact on speed. It will increase the time it takes to load after running a command, which could be significant going from 30 columns to 100 columns on slower machines. We could test this out.

2. The unresolved crashing of RDotNet displaying **some** wide datasets. We didn't get to the bottom of this but I think this was part of the original reason for 30 columns which seemed not to give the problem even on the problem datasets. It might be rare enough that we are ok to increase the number but it does cause a total crash of the software.

We might want to try increasing the value for the development team first and see what the effects are before deciding.

**Comment by rdstern** [comment]
I hope the 2021 code sprint can allow an investigation of this issue.
Currently the limits are 1000 cases by 30 variables.  

There are 3 linked parts to the investigation:
a) Should the default limits be changed with climatic applications in mind?  In my climatic work 30 variables is sometimes limiting.  Even 40 would be better and 100 would be more than I need.  In cases (rows) I usually have many more than 1000.  Changing to (say) 2500 cases would sometimes be helpful, but not particularly.  We often have 10,000 to 100,000 rows and it can be more.
b) In other applications there can be many more variables.  One example to be used during the sprint has 330 variables and the procurement data from the World Bank - in the library - has almost 200.  Again the survey data used in the sprint has over 50,000 cases.

In summary, in this investigation, I would like a bit more of the emphasis to be on the impact of increasing the number of variables.

The related issue is that the user is free to change the limits for themselves.  There is a natural temptation - as in Excel - to make the whole data frame visible.  For example 330 variables by 60,000 cases is enough for the survey data.  We currently give no warnings that this might be a bad idea!  In Tools > Options in the relevant tab there is ample space to provide guidance.  What should we say?

**Comment by Karakacha** [comment]
Viewing variable names is very key since a user does not have to master all variable names and it can be better if we can increase  this number to the extend that it can accommodate most variable names for wider data set.  Perhaps we may think of increasing it to 100 and also increase it later to be even accommodating all variables of the data set despite its number in future versions. I will strongly support increase of variable names.

**Comment by dannyparsons** [comment]
I don't think we should increase the row limit, that makes running slower and I don't think we have a a very good reason to.

On columns I can see the advantage, but we had instability with this before. So we need to test it very carefully. I repeat my suggestion below: I would not be comfortable changing the default without extensive testing in case it causes crashes for lots of users.

> We might want to try increasing the value for the development team first and see what the effects are before deciding.

**Comment by Karakacha** [comment]
Definitely variables are more important than rows. It can be better to
reduce cases but accommodate most variables so as to maintain the speed of
importing the dataset. Hope to see you have a meaningful discussion about
it with the developers.

**Comment by rdstern** [comment]
@Karakacha the idea wasn't just to say what you would like.  It was to do careful tests of what happens.  That's one reason I wanted you to examine the Pakistan data, namely to be able to see if increasing the number of variables makes things run more slowly - how much more slowly?  Can you try and then extrapolate?

And, with also the possibility that it might crash - which would be a disaster, given that it now seems more stable.

Also I am concerned that we have a low limit 1000 by 30 which is good and seems stable.  But in the Tools Options we permit a user to make the value whatever they like - and we don't currently have any advice from our own experience.  For example what does happen if you do what (to me) seems intuitively reasonable - as it is roughly like a spreadsheet, then (for the Pakistan data, I'll increase to 300 variables and 60,000 rows.  I wonder what does happen?

**Comment by Karakacha** [comment]
Kindly attach to me the Pakistan data so that I May test this

**Comment by rdstern** [comment]
Look at #6289 under the discussions tab, to see how to download these MIPS data, or ask @TobbyKe or @jkmusyoka who may have it already

**Comment by jkmusyoka** [comment]
@Karakacha I had already downloaded the Pakistan dataset and would be happy to pass it over for the testing.

**Comment by TobbyKe** [comment]
Punjab- Pakistan Data
 Pakistan Punjab MICS6 Datasets.zip
<https://drive.google.com/file/d/1hUThcHBUdblRI2YvAIUpKZxjLOOe3nKG/view?usp=drive_web>

**Comment by Karakacha** [comment]
The default value of 30 variables and 1000 cases is very stable and works
efficiently. importing data is also quick and data viewer is simple to
follow

 Perhaps this may not be suitable to a user who has more than 100 variables
and thousands of cases. I had to test this using the HH dataset from
Pakistan to see if R-Instat can support. I realized that you can
increase your variables to 100 and 10,000 cases and be imported
successfully. I increased further to 330 variables and 10,000 cases, it
took quite a while but it was imported. I had also to do some analyses at
this stage and it was still Ok. Further increase to 15,000 cases, the
software stopped responding.

If we may stick to our default of 1000 cases,( which I also support) it is
convenient since R-Instat is stable when you import maximum variables
(maybe beyond 330) and carry out the analysis. If there may be that concern
of increasing variables it is fine for the software supports it.

**Comment by rdstern** [comment]
@ChrisMarsh82 we missed Version 0.7.5, which would have been useful, in both climatic workshops, where we had columns for each day of the month, so more than 30 variables - though less than 40.
So I am still happy with the number of rows remaining at 1000.  Could we please raise the (default) number of variables.  I think you said 100 would be possible, that thet would be great.  But, if that is too many, then please consider 50, or 40 at least.

**Comment by Patowhiz** [comment]
@rdstern can you confirm that 50 columns is ideal, or do a test with 100 displayed columns using a long data sets. If it hold well, I can increase the default to 100 or the 330 suggested above. Will be good to strike a balance on the sweet spot.

**Comment by rdstern** [comment]
@Patowhiz I would be happy to be cautious and have 1000 rows and 50 variables for now.  Then also to have the development team routinely using 1000 rows and 100 variables for now, and possibly checking whether there are speed differences?

I don't think we need more than 100, as I really like the paging that @ChrisMarsh82 has provided.  So having round numbers makes it more obvious what you are paging to, in either rows or columns.  That's why I like 50 better than 30 - and 100 would be even simpler.  But I would prefer to be cautious for now - hence the suggestion of 50.
What I would also like is the feature that I think @N-thony is waiting for you to do a custom control on, namely to be able to jump to a specified "page" - either across or down.  Currently if you have 100,000 rows (or 7,000 variables) and you want to get to row 50,000+ (or variables 3500 plus) then you have to step one at a time.  I am quite happy that it sticks to defined pages, but would like to jump to a particular one.
